src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,51 @@
 package core.basesyntax.service;
 
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
+import java.util.Objects;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new NullPointerException("User can't be register");
+        }
+        if (user.getLogin() == null) {
+            throw new NullPointerException("Invalid login");
+        }
+        if (user.getPassword() == null) {
+            throw new NullPointerException("Invalid password");
+        }
+        if (user.getAge() == null) {
+            throw new NullPointerException("Invalid age");
+        }
+        checkingAge(user);
+        checkingPasswordLength(user);
+        checkingInStorage(user);
```

Comment:
no need to separate validation logic into 3 different methods, one if statement with checks will do
also there is get method in storageDao, you can use it to check if user already exists in storage